AnonUpload/src/main/java/com/theironyard/controllers/AnonFileController.java

```diff
@@ -24,18 +26,35 @@
     @Autowired
     AnonFileRepository repository;
 
+    public static final int LIMIT = 5;
+
     @RequestMapping(path = "/upload", method = RequestMethod.POST)
-    public void upload(MultipartFile file, HttpServletResponse response) throws IOException {
+    public void upload(MultipartFile file, HttpServletResponse response, String permanent, String comment,String password) throws IOException, PasswordStorage.CannotPerformOperationException {
         File dir = new File("public/files");
         if(!dir.exists()) {
             dir.mkdirs();
         }
-
         File f = File.createTempFile("anonfile", file.getOriginalFilename(), dir);
         FileOutputStream fos = new FileOutputStream(f);
         fos.write(file.getBytes());
-
         AnonFile anonFile = new AnonFile(f.getName(), file.getOriginalFilename());
+        if(anonFile.getPassword() == null){
```

Comment:
How is this intended to work?  If password is null then set the password to a password hash of null?